lib/instrumentation/modules/amqplib.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+var shimmer = require('../shimmer')
+var { getDBDestination } = require('../context')
+
+var URL = require('url-parse')
+
+module.exports = function (amqplib, agent, { version, enabled }) {
+  agent.logger.debug('shimming amqplib.connect')
+  shimmer.wrap(amqplib, 'connect', wrapConnect)
+
+  return amqplib
+
+  function patchChannelModel (channelModel, host, port) {
+    agent.logger.debug('shimming amqplib.ChannelModel.connection.sendMessage')
+    shimmer.wrap(channelModel.connection, 'sendMessage', function (orig) {
+      return function (channel, method, fields, properties, props, content) {
+        var span = enabled && agent.startSpan(null, 'messaging', 'amqp', 'publish')
```

Comment:
shouldn't `publish` be the actual method that gets called?

Reply:
`sendMessage` is the method needed to be patched but I've thought that `publish` is more universal. That's why I've named the span like this. Are you ok with that?